### PR TITLE
Subgraph and Deployer Contracts

### PR DESCRIPTION
**Pull request by daodesigner**
Squash and merge package into monorepo, integrate linter, new tests. Need to push back the failing tests from @pettinarip's branch once they are passing (CI blocks pushing otherwise).

- [x] There are unit tests on the `subgraph` branch that seem to be failing, need to straighten this out before merge
- [x] Replace the unit tests in `vue-app/tests/unit` and `contracts/tests/recipientRegistry.ts` that are blocking CI

### CODE REVIEW

File: subgraph/src/FundingRoundFactoryMapping.ts [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+  fundingRound.matchingPoolSize = matchingPoolSize
+
+  //Check if these registries already exist/are being tracked
+  let recipientRegistryAddress = fundingRoundFactoryContract.recipientRegistry()
+  let recipientRegistryId = recipientRegistryAddress.toHexString()
+  let recipientRegistry = RecipientRegistry.load(recipientRegistryId)
+
+  let contributorRegistryAddress = fundingRoundFactoryContract.userRegistry()
+  let contributorRegistryId = contributorRegistryAddress.toHexString()
+  let contributorRegistry = ContributorRegistry.load(recipientRegistryId)
```
Comment (pettinarip) [inline subgraph/src/FundingRoundFactoryMapping.ts]:
```suggestion
  let contributorRegistry = ContributorRegistry.load(contributorRegistryId)
```

File: subgraph/subgraph.yaml [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+          handler: handleContribution
+        - event: ContributionWithdrawn(indexed address)
+          handler: handleContributionWithdrawn
+        - event: FundsClaimed(indexed uint256,indexed address,uint256)
+          handler: handleFundsClaimed
+        - event: OwnershipTransferred(indexed address,indexed address)
+          handler: handleOwnershipTransferred
+        - event: TallyPublished(string)
+          handler: handleTallyPublished
+      callHandlers:
```
Comment (pettinarip) [inline subgraph/subgraph.yaml]: The `callHandlers` were giving me `trace_filter` errors due that HH do not support that. Wondering if there is a different way of doing it (at least for the local development, not a high priority) or a way we can omit that in local dev 🤔

File: vue-app/tests/unit/project-list-item.ts [hunk 3]
```diff
@@ -1,16 +1,16 @@
...
+//     const wrapper = shallowMount(ProjectListItem, {
+//       propsData: { project: { description: '' } },
+//       mocks: {
+//         $store: { state: { cart: [] } },
+//       },
+//       stubs: ['router-link'],
+//     })
+//     expect(wrapper.text()).to.include('')
+//   })
+// })
```
Comment (pettinarip) [inline vue-app/tests/unit/project-list-item.ts]: I think you've already fixed your troubles with these tests. We can enable them again.

**Review by pettinarip — COMMENTED** [review]
Looking good to me Q. I left a couple of minor comments.

File: subgraph/subgraph.yaml (hunk 2 above)
Comment (daodesigner) [reply subgraph/subgraph.yaml]: removed it

File: vue-app/tests/unit/project-list-item.ts (hunk 3 above)
Comment (daodesigner) [reply vue-app/tests/unit/project-list-item.ts]: added tests back in, they fail CI without a valid .env file

**Review by pettinarip — APPROVED** [review]

**Pull request merged**